### PR TITLE
Various edits of developer guide

### PR DESCRIPTION
**Pull request by kwankyu**
Mostly small edits and 

- updated sections on testing facilities
- added how to go back exited container.
- moved the section "Testing GitHub Actions locally" to an appropriate place.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kwankyu** [comment]
@mkoeppe 

I created this PR draft since I experienced a certain failure in following "Testing on Multiple Platforms" of the developer guide.  Precisely I followed the guide in the section "Testing on multiple platforms using Docker", and created a docker container with "fedora:latest" image:
```
docker run -it --mount type=bind,source=$(pwd),target=/sage fedora:latest
```
but during "make -j4", I got lots of errors like
```
...
[python3-3.11.1] cp: preserving permissions for '/sage/local/var/lib/sage/venv-python3.11.1/./include/python3.11/weakrefobject.h': No data available
[python3-3.11.1] cp: preserving permissions for '/sage/local/var/lib/sage/venv-python3.11.1/./include/python3.11/pyconfig.h': No data available
[python3-3.11.1] cp: preserving permissions for '/sage/local/var/lib/sage/venv-python3.11.1/./share/man/man1/python3.11.1': No data available
```
Hence I suspect that there is something wrong in the guide. By the way, there is no problem in building sage in a container with no `--mount type=bind,source=$(pwd),target=/sage` (of course, in this case, I created `/sage` directory as a sage git repository).

Do you know what is wrong in the guide?

**Comment by mkoeppe** [comment]
As far as I can see, the command is correct as is; compare with https://docs.docker.com/storage/bind-mounts/#start-a-container-with-a-bind-mount

Try if you see the same problem with a different Docker image. 
(I can't try it right now.)

**Comment by kwankyu** [comment]
Right. There's no problem with ubuntu and archlinux images. Thus it seems fedora problem...

**Comment by kwankyu** [comment]
Another question: are we still running sage buildbots? If no, the section on "sage buildbots" could be removed.

**Comment by kwankyu** [comment]
sage build succeeded in the ubuntu container, but in archlinux container I got again
```
[singular-4.3.2p8] cp: preserving permissions for '/sage/local/./bin/libpolys-config': No data available
[singular-4.3.2p8] cp: preserving permissions for '/sage/local/./bin/Singular': No data available
[singular-4.3.2p8] cp: preserving permissions for '/sage/local/./bin/ESingular': No data available
[singular-4.3.2p8] cp: preserving permissions for '/sage/local/./bin/TSingular': No data available
[singular-4.3.2p8] cp: preserving permissions for '/sage/local/./bin/libparse': No data available
[singular-4.3.2p8] cp: preserving permissions for '/sage/local/./bin/libsingular-config': No data available
[singular-4.3.2p8] ************************************************************************
[singular-4.3.2p8] Error moving files for singular-4.3.2p8.
[singular-4.3.2p8] ************************************************************************
[singular-4.3.2p8] Please email sage-devel (http://groups.google.com/group/sage-devel)
[singular-4.3.2p8] explaining the problem and including the log files
[singular-4.3.2p8]   /sage/logs/pkgs/singular-4.3.2p8.log
[singular-4.3.2p8] and
[singular-4.3.2p8]   /sage/config.log
[singular-4.3.2p8] Describe your computer, operating system, etc.
[singular-4.3.2p8] ************************************************************************
make[4]: *** [Makefile:3380: singular-SAGE_LOCAL-no-deps] Error 1
make[3]: *** [Makefile:3380: /sage/local/var/lib/sage/installed/
```
Strange...

**Comment by mkoeppe** [comment]
> Another question: are we still running sage buildbots?

"We" are, as in, Volker runs them as part of what he's doing.

File: src/doc/en/developer/portability_testing.rst [hunk 1]
```diff
@@ -10,40 +10,44 @@ Testing on Multiple Platforms
...
-will automatically test your GitHub PR by attempting an incremental build of
-Sage and running doctests.
+Testing PRs with GitHub Actions
+===============================
+
+`GitHub Actions <https://github.com/sagemath/sage/actions>`_ are automatically
+and constantly testing GitHub PRs to identify errors early and ensure code
+quality. In particular, Build & Test workflows perform an incremental build of
+Sage and run doctests on a selection of major platforms including Ubuntu,
+MacOS, and Conda.
```
Comment (mkoeppe) [inline src/doc/en/developer/portability_testing.rst]:
```suggestion
macOS, and Conda.
```

Comment (kwankyu) [reply src/doc/en/developer/portability_testing.rst]: Thanks. Done.

**Pull request merged**